### PR TITLE
registry: index manifest artifact type for filtering (PROJQUAY-7471)

### PR DESCRIPTION
**Pull request by kleesc**
Previous assumption made use of the config media type only, which is not the case if a manifest's artifact type is explicitly set. i.e the config's media type and artifact type are different, and the artifact type take precedence for filtering.

### CODE REVIEW

File: data/migrations/versions/3f8d7acdf7f9_add_artifact_type_column.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""Add artifact_type column
+
+Revision ID: 3f8d7acdf7f9
+Revises: 2664723e1b4b
```
[the remark this answers is not in the document]
Comment (kleesc) [reply data/migrations/versions/3f8d7acdf7f9_add_artifact_type_column.py]: fixed

File: data/migrations/versions/3f8d7acdf7f9_add_artifact_type_column.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    manifest_indexes = inspector.get_indexes("manifest")
+
+    if not "artifact_type" in [c["name"] for c in manifest_columns]:
+        op.add_column("manifest", sa.Column("artifact_type", sa.String(length=255), nullable=True))
+
+    if not "artifact_type_backfilled" in [c["name"] for c in manifest_columns]:
+        op.add_column("manifest", sa.Column("artifact_type_backfilled", sa.Boolean()))
+
+    if not "manifest_repository_id_artifact_type" in [i["name"] for i in manifest_indexes]:
+        op.create_index(
```
Comment (bcaton85) [inline data/migrations/versions/3f8d7acdf7f9_add_artifact_type_column.py:31]: we'll need the `with op.get_context().autocommit_block():` here too

**Review by bcaton85 — DISMISSED** [review]
LGTM

**Review by Sunandadadi — APPROVED** [review]
lgtm

**Review by bcaton85 — APPROVED** [review]
LGTM

**Pull request merged**